### PR TITLE
modularize helper functions and add pytests

### LINKED ISSUE #88: UnitTest
Choose any of the following formats:

- Unit tests (pytest): test functions in isolation with explicit inputs and expected outputs.
- Doctests: embed examples directly in your function docstrings.
- Analysis notebook (notebooks/logic_tests.ipynb): run your main filtering or aggregation functions across representative parameter combinations and show expected vs. actual output.

Regardless of format, your tests must:

- Cover at least 3 distinct behaviors using playwright (e.g., edge-case filter, aggregation correctness, a data type or boundary condition).
- Refactor your code into at least 1 function and write a unit test (pytest) for it.
- Include a one-sentence description per test explaining what behavior is being verified and why it matters for the dashboard.
- Pass on a clean environment. Document how to run them in your README.
- Tests must be runnable with a single command documented in your README.
- In your reflection, describe what each test covers and what could break if the behavior changes.

### PR DESCRIPTION
**Pull request by j031nich0145**
In this PR:

- refactors app.py creating separate modules for 2 helper functions: fmt_num, fmt_currency
- creates test files for the helper functions in a /tests directory with pytest
- multiple tests per function with all passing
-  added init.py file to source to be treated as a module
- added pytest.ini file
-  added 3 playwrite tests for live dashboard
- added pytest and playwrite to environment.yml file
- updated changelog

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -63,19 +61,19 @@
...
+#     if v >= 1e12:  return f"{s}${v/1e12:.2f}T"
+#     if v >= 1e9:   return f"{s}${v/1e9:.1f}B"
+#     if v >= 1e6:   return f"{s}${v/1e6:.1f}M"
+#     if v >= 1e3:   return f"{s}${v/1e3:.1f}K"
+#     return f"{s}${v:.0f}"
+
+# def fmt_num(v):
+#     if v >= 1e6:  return f"{v/1e6:.1f}M"
+#     if v >= 1e3:  return f"{v/1e3:.1f}K"
+#     return f"{v:,.0f}"
```
Comment (clr-saunders) [inline src/app.py:66]: Thanks Joel! I think we can remove the commented out code :).

File: tests/test_fmt_num.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+from src.fmt_num import fmt_num
+
+def test_millions():
+    assert fmt_num(2_500_000) == "2.5M"
+
+def test_thousands():
+    assert fmt_num(18_000) == "18.0K"
+
+def test_thousands():
```
Comment (clr-saunders) [inline tests/test_fmt_num.py:13]: Looks like a duplicate test snuck in. And we'll need the sentence description per test for this file as well, explaining what behaviour is being verified and why it matters for the dashboard.

File: tests/test_fmt_currency.py
Comment (clr-saunders) [inline tests/test_fmt_currency.py:1]: Nice work with unit test Joel. To meet milestone requirements we need t  "Include a one-sentence description per test explaining what behavior is being verified and why it matters for the dashboard."

**Review by clr-saunders — CHANGES_REQUESTED** [review]
Hi Joel — thanks for the work on this! The helper refactor and pytest unit tests look good. A couple items are still needed to fully meet the M4 rubric:

1. **Playwright tests:** The rubric requires tests covering *at least 3 distinct behaviors using Playwright* (e.g., filter behavior, edge cases, aggregation correctness). The current tests are pytest unit tests only, so we’ll still need the Playwright UI tests.

2. **Test descriptions:** Each test should include a **one-sentence description** explaining what behavior is being verified and why it matters for the dashboard.

3. **Environment dependencies:** `pytest` and `playwright` should be added to `environment.yml` so the tests run in a clean environment.

4. **Minor cleanup:**

   * Remove the commented-out helper code in `app.py`.
   * `test_fmt_num.py` defines `test_thousands()` twice (one will overwrite the other).

Once those are added this should satisfy the testing portion of the milestone.

File: tests/test_fmt_num.py (hunk 2 above)
Comment (j031nich0145) [reply tests/test_fmt_num.py:13]: Good catch Claire, thanks!

**Comment by j031nich0145** [comment]
Thanks for the attention to the details, Claire - yikes no more labs at 1am for me :P 
I swear I had definitely at least deleted the commented out helper functions, don't know what happened there.

**Comment by clr-saunders** [comment]
No worries at all! that's what reviews are for - more eyes so we can all catch these things, I hear you, it's so easy to miss details especially late at night with lots of labs on the go.  I actually think we all missed the playwright detail when we were looking at it in lab. All good! :)

**Comment by j031nich0145** [comment]
There are some minor merge conflicts in the environment file I had to update, as well as the app.py which I required to update the chatbot api key in as well

**Review by clr-saunders — APPROVED** [review]
Nice, thanks Joel! I tested everything by rebuilding the environment from your `environment.yml` branch.

One thing I noticed: running `shiny run src/app.py` threw an import error, but it works with:

PYTHONPATH=. shiny run src/app.py

So I’ll update the README in my README branch to use that command. After that I ran `pytest` and all tests passed 🎉 Good to merge once the merge conflicts are resolved. I'll resolve them now and then you are good to merge this on in :).

**Comment by j031nich0145** [comment]
Amazing, thanks Claire!
